src/ifcopenshell-python/ifcopenshell/ids.py

```diff
@@ -52,12 +52,15 @@ def __init__(self, node):
         self.node = node
 
     def __getattr__(self, k):
-        v = self.node.getElementsByTagName(k)[0]
-        elems = [n for n in v.childNodes if n.nodeType == n.ELEMENT_NODE]
-        if elems:
-            return restriction(elems[0])
-        else:
-            return v.firstChild.nodeValue.strip()
+        try:
+            v = self.node.getElementsByTagName(k)[0]
```

Comment:
I guess the `IndexError` is only about the first line `self.node.getElementsByTagName(k)[0]`. My preference then would be to only try-catch that line so that we don't by accident hide more errors later.

Reply:
ok, I narrowed it down to that particular line.